CMake/External_GTest.cmake

```diff
@@ -0,0 +1,23 @@
+ExternalProject_Add(GTest
+  URL ${GTest_url}
+  URL_MD5 ${GTest_md5}
+  PREFIX ${fletch_BUILD_PREFIX}
+  DOWNLOAD_DIR ${fletch_DOWNLOAD_DIR}
+  INSTALL_DIR ${fletch_BUILD_INSTALL_PREFIX}
+  PATCH_COMMAND ${CMAKE_COMMAND}
+    -DGTest_patch:PATH=${fletch_SOURCE_DIR}/Patches/GTest
+    -DGTest_source:PATH=${fletch_BUILD_PREFIX}/src/GTest
+    -P ${fletch_SOURCE_DIR}/Patches/GTest/Patch.cmake
+  CMAKE_GENERATOR ${gen}
+  CMAKE_ARGS
+    ${COMMON_CMAKE_ARGS}
+  )
+
+set(GTEST_ROOT ${vision-tpl_BUILD_INSTALL_PREFIX} CACHE STRING "")
+
+file(APPEND ${vision-tpl_CONFIG_INPUT} "
+########################################
+# GTest
+########################################
+set(GTEST_ROOT    @GTEST_ROOT@)
```

Comment:
Fletch has a new pattern for writing out this file. See one of the simple cases, e.g. ExternalZLib.cmake. In short, it looks something like...

file(APPEND ${fletch_CONFIG_INPUT} "
set(GTEST_ROOT \$\{fletch_ROOT\})
set(fletch_ENABLED_GTEST_ROOT TRUE)
")